java-extras/src/main/java/org/triplea/java/function/NegationPredicate.java

```diff
@@ -0,0 +1,27 @@
+package org.triplea.java.function;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.function.Predicate;
+
+/**
+ * Utility for a 'not' predicate, ie: a test that a boolean expression is false.
+ */
+public final class NegationPredicate {
+  private NegationPredicate() {
+
+  }
+
+  /**
+   * Returns a predicate that represents the logical negation of the specified predicate.
+   *
+   * @param p The predicate to negate.
+   *
+   * @return A predicate that represents the logical negation of the specified predicate.
+   */
+  public static <T> Predicate<T> not(final Predicate<T> p) {
```

Comment:
I get the impression that you view a 'class' as a high cost. That reminds me of Java 1.2 and 1.3 when classes *were* **very** expensive and clever tricks were used to avoid them (and to good effect). As an anecdote, I was checking Java perf and tested a no-op method vs a new class in Java 1.8, the new class was at times faster faster than a no-op method, statistically the difference was not significant.

If we are concerned about an explosion of many micro-classes, in this case the package scope is pretty well defined and I do not think we would actually have that problem until we added another dozen or more classes. We have some breathing room.

To another extent, the original authors/designers of the existing code decided already that the method should not be a local utility and had it extracted to its own stand-alone class, ie: `Util.java`. From that perspective the decision was already made to have this in its own class.

`Util.java` has a number of problems to it. @RoiEXLab would you agree we should try to deprecate `Util.java` and break-up and/or move its methods to more appropriate locations? Noting the TODOs on the` Util.java`, all of its methods are slated to move, it will no longer exist. At some point we could have the problem that this would be the only method on `Util.java`.

Reply:
I did think about this today and I came to the conclusion that my point of view is just based on the impression that splitting up content into multiple classes just seems like it turns everything into "more" code.
On the other hand I now think splitting up classes is a useful thing to do, when you keep class loading in mind. The more classes are split into logical groups, the less unecessary methods are being loaded.
So no objections.